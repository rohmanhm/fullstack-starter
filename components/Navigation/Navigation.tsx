import { NAVIGATION_MENU } from '@/configs'
import { Container, Flex, useColorModeValue as mode } from '@chakra-ui/react'
import { useRouter } from 'next/router'
import { FC } from 'react'
import { NavigationMenu } from './NavigationMenu'

const Navigation: FC = () => {
  const router = useRouter()
  return (
    <Flex
      w="100%"
      position="fixed"
      bottom="0"
      left="0"
      borderTop="1px solid"
      borderColor={mode('gray.200', 'gray.600')}
      color="gray.500"
    >
      <Container maxW="container.md" display="inline-flex" px="0">
        {NAVIGATION_MENU.map((menu, i) => (
          <NavigationMenu
            key={i}
            isActive={menu.route === router.pathname}
            {...menu}
          />
        ))}
      </Container>
    </Flex>
  )
}

export default Navigation
