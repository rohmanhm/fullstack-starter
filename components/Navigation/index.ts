export * from './Navigation'
export * from './NavigationMenu'
export * from './withNavigation'
