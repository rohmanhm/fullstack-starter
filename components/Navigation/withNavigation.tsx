import { ComponentType, FC } from 'react'
import Navigation from './Navigation'

export const withNavigation = (Component: ComponentType) => {
  const enhanced: FC = (props) => {
    return (
      <>
        <Component {...props} />
        <Navigation />
      </>
    )
  }

  enhanced.displayName = 'WithNavigation'

  return enhanced
}
