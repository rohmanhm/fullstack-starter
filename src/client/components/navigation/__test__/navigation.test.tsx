import { render } from '@testing-library/react'
import React from 'react'
import { HiCollection } from 'react-icons/hi'
import Navigation, { NAVIGATION_MENU_TEST_ID } from '../navigation'

const navItems = [
  {
    icon: HiCollection,
    label: 'Menu',
    pathname: '/home',
  },
]

describe('Navigation Menu', () => {
  it('matches snapshot', () => {
    const emptyItems = render(<Navigation />)
    expect(emptyItems.asFragment()).toMatchSnapshot()

    const { asFragment } = render(<Navigation items={navItems} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('render items correctly', () => {
    const { getAllByTestId } = render(<Navigation items={navItems} />)
    expect(getAllByTestId(NAVIGATION_MENU_TEST_ID)).toHaveLength(
      navItems.length
    )
  })
})
