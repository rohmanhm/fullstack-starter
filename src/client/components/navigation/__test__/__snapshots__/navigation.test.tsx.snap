// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Menu matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-12htcqv"
  >
    <div
      class="chakra-container css-je61vd"
    />
  </div>
</DocumentFragment>
`;

exports[`Navigation Menu matches snapshot 2`] = `
<DocumentFragment>
  <div
    class="css-12htcqv"
  >
    <div
      class="chakra-container css-je61vd"
    >
      <a
        class="css-13cu51q"
        data-testid="navigation-menu"
      >
        <svg
          class="chakra-icon css-f0a3g4"
          fill="currentColor"
          focusable="false"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 20 20"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7 3a1 1 0 000 2h6a1 1 0 100-2H7zM4 7a1 1 0 011-1h10a1 1 0 110 2H5a1 1 0 01-1-1zM2 11a2 2 0 012-2h12a2 2 0 012 2v4a2 2 0 01-2 2H4a2 2 0 01-2-2v-4z"
          />
        </svg>
        <p
          class="chakra-text css-1wrp73i"
        >
          Menu
        </p>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
